### PR TITLE
需要增加请求charset非空判断，避免charset.name()空指针异常#1403 2

### PR DESCRIPTION
**Pull request by JiangZian**
需要增加请求charset非空判断，避免charset.name()空指针异常#1403 2

### CODE REVIEW

**Comment by sczyh30** [comment]
Hi, thanks for contributing. Could you please [sign the CLA here](https://cla-assistant.io/alibaba/Sentinel?pullRequest=1587)? And please make sure the email of your commits match your GitHub email. You may refer to the instruction [here](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user), or rebase your commits with the correct email.

---

感谢贡献，请将 commit 对应的 email 调整成与 GitHub 的 email 相匹配并 [确认一下 CLA](https://cla-assistant.io/alibaba/Sentinel?pullRequest=1587)。

File: sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/client/SimpleHttpClient.java [hunk 1]
```diff
@@ -152,10 +155,14 @@ private String getStatusLine(RequestMethod type, String requestPath) {
...
      *
      * @param paramsMap pair of parameters
-     * @param charset   charset
+     * @param charset   charset is not allowed to be null
      * @return encoded request parameters, or empty string ("") if no parameters are provided
      */
-    private String encodeRequestParams(Map<String, String> paramsMap, Charset charset) {
+    private String encodeRequestParams(Map<String, String> paramsMap, Charset charset) throws RequestException{
+        if (charset == null){
+            CommandCenterLog.warn("Illegal request: null charset");
```
Comment (sczyh30) [inline sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/client/SimpleHttpClient.java:163]: Maybe this is not needed, as the upper layer will catch and record it.

**Comment by JiangZian** [comment]
您好，感谢您的回复
您说的前两点我是非常赞同的，
但是第三点必要性，charset 空的msg
我认为还是要明确返回信息的，这是一个比较重要的因子，在流程逻辑中，没有明确的null值日志
------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Eric Zhao"<notifications@github.com&gt;;
发送时间:&nbsp;2020年7月6日(星期一) 晚上6:01
收件人:&nbsp;"alibaba/Sentinel"<Sentinel@noreply.github.com&gt;;
抄送:&nbsp;"江子涵"<576811031@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [alibaba/Sentinel] 需要增加请求charset非空判断，避免charset.name()空指针异常#1403 2 (#1587)





 
@sczyh30 commented on this pull request.
 
 
In sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/client/SimpleHttpClient.java:
 &gt;       * @return encoded request parameters, or empty string ("") if no parameters are provided       */ -    private String encodeRequestParams(Map<String, String&gt; paramsMap, Charset charset) { +    private String encodeRequestParams(Map<String, String&gt; paramsMap, Charset charset) throws RequestException{ +        if (charset == null){  
Could you please reformat your code?
 
 
In sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/client/SimpleHttpClient.java:
 &gt;       * @return encoded request parameters, or empty string ("") if no parameters are provided       */ -    private String encodeRequestParams(Map<String, String&gt; paramsMap, Charset charset) { +    private String encodeRequestParams(Map<String, String&gt; paramsMap, Charset charset) throws RequestException{ +        if (charset == null){ +            CommandCenterLog.warn("Illegal request: null charset"); +            throw new RequestException(StatusCode.BAD_REQUEST, "charset is not allowed to be null");  
RequestException and CommandCenterLog are designed for command center. Maybe here we could simply use IllegalArgumentException.
 
 
In sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/client/SimpleHttpClient.java:
 &gt;       * @return encoded request parameters, or empty string ("") if no parameters are provided       */ -    private String encodeRequestParams(Map<String, String&gt; paramsMap, Charset charset) { +    private String encodeRequestParams(Map<String, String&gt; paramsMap, Charset charset) throws RequestException{ +        if (charset == null){ +            CommandCenterLog.warn("Illegal request: null charset");  
Maybe this is not needed, as the upper layer will catch and record it.

File: sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/client/SimpleHttpClient.java (hunk 1 above)
Comment (sczyh30) [reply sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/client/SimpleHttpClient.java:163]: > 但是第三点必要性，charset 空的msg，我认为还是要明确返回信息的，这是一个比较重要的因子，在流程逻辑中，没有明确的null值日志.

This has been caught and recorded in upper `SimpleHttpHeartbeatSender`:

https://github.com/alibaba/Sentinel/blob/5e1a44366303920589b171cf08c9278bb1671094/sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/SimpleHttpHeartbeatSender.java#L74-L84

So we may just carry the message in IllegalArgumentException.

Comment (JiangZian) [reply sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/client/SimpleHttpClient.java:163]: 好的，感谢您的解答

**Comment by JiangZian** [comment]
Due to the email problem, the operation such as submission was carried out again

**Pull request closed without merging**